src/FakeItEasy/OutputWriterExtensions.cs

```diff
@@ -49,5 +51,72 @@ public static IOutputWriter Write(this IOutputWriter writer, object value)
             writer.Write(value.ToString());
             return writer;
         }
+
+        /// <summary>
+        /// Formats the specified argument values as strings and writes
+        /// them to the output, separated by the specified separator.
+        /// </summary>
+        /// <param name="writer">The writer to write to.</param>
+        /// <param name="values">The values to write to the writer.</param>
+        /// <param name="separator">The separator used to separate the values.</param>
+        /// <param name="skipFormatting">Specify that argument values should not be formatted.</param>
+        /// <returns>The writer.</returns>
+        internal static IOutputWriter WriteArgumentValues(this IOutputWriter writer, IEnumerable values, string separator, bool skipFormatting = false)
+        {
+            Guard.AgainstNull(writer, nameof(writer));
+            Guard.AgainstNull(values, nameof(values));
+
+            bool first = true;
+            foreach (var value in values)
+            {
+                if (!first)
+                {
+                    writer.Write(separator);
+                }
+
+                if (skipFormatting)
+                {
+                    writer.Write(value);
+                }
+                else
+                {
+                    writer.WriteArgumentValue(value);
+                }
+
+                first = false;
+            }
+
+            return writer;
+        }
+
+        /// <summary>
+        /// Formats the specified argument values as strings and writes
+        /// them to the output, separated by the specified separator.
+        /// </summary>
+        /// <param name="writer">The writer to write to.</param>
+        /// <param name="values">The values to write to the writer.</param>
+        /// <param name="separator">The separator used to separate the values.</param>
+        /// <param name="maxNumber">The maximum number of elements to write.</param>
+        /// <param name="skipFormatting">Specify that argument values should not be formatted.</param>
+        /// <returns>The writer.</returns>
+        internal static IOutputWriter WriteArgumentValues(this IOutputWriter writer, IEnumerable values, string separator, int maxNumber, bool skipFormatting = false)
+        {
+            Guard.AgainstNull(writer, nameof(writer));
+            Guard.AgainstNull(values, nameof(values));
+
+            var list = values.AsList();
+            if (list.Count <= maxNumber)
+            {
+                writer.WriteArgumentValues(list, separator, skipFormatting);
+            }
+            else
+            {
+                writer.WriteArgumentValues(list.Take(maxNumber - 1), separator, skipFormatting);
+                int remainingCount = list.Count - maxNumber + 1;
+                writer.Write($", ... ({remainingCount} more elements)");

```

Comment:
> Should `,` not be the passed-in separator?

Yes. Although I doubt we'll ever use anything other than `,`... we could probably remove the parameter and hard-code it instead.

> Perhaps we could print all the elements if less than six, otherwise the first two and last two?

👍

Reply:
I think removing the parameter is a good plan.

Reply:
While I was at it, I also removed the `maxNumber` parameter.